src/bundles/config.js

```diff
@@ -13,6 +17,15 @@ const bundle = createAsyncResourceBundle({
       conf = JSON.stringify(rawConf, null, '\t')
     }
 
+    // More info: https://github.com/ipfs-shipyard/ipfs-webui/issues/1490#issuecomment-671633602
+    const config = JSON.parse(conf)
+    const url = getURLFromAddress('Gateway', config) || DEFAULT_URI
+
+    if (!await checkIfGatewayUrlIsAccessible(url)) {
+      config.Addresses.Gateway = DEFAULT_MULTI_ADDR
+      return JSON.stringify(config)
```

Comment:
After giving it a spin it works fine on _Files_ and _Explore_ screens, but I don't think this will work for _Settings_.

When `if (true)`  you will see that it breaks IPFS Config on _Settings_ page:

> ![2020-08-20--10-13-27](https://user-images.githubusercontent.com/157609/90736294-f7b18200-e2cd-11ea-9444-7000d45d9dba.png)

We don't want `Addresses.Gateway` to be changed on the _Settings_ page because if user edits and submits new config that sets `Addresses.Gateway = /dns4/ipfs.io/tcp/443/https`, which will break gateway functionality of their node.

We could either:
- (A) not modify original `config`, but create separate value/bundle to represent "available gateway" and switch _Files_ and _Explore_ to use that value if present (the cleanest solution)
- (B) create separate bundle for _Settings_ screen which does not modify `Addresses.Gateway` (ugly fix)

@rafaelramalho19  I'd got with (A) unless you see a better way

Reply:
Good find! Thank you, fixed 👍